crates/vsock/src/main.rs

```diff
@@ -287,116 +287,182 @@ mod tests {
     }
 
     #[test]
-    #[serial]
     fn test_vsock_config_setup() {
-        let args = VsockArgs::from_args(3, "/tmp/vhost4.socket", "/tmp/vm4.vsock", 64 * 1024);
+        let test_dir = tempdir().expect("Could not create a temp test directory.");
+        std::env::set_current_dir(&test_dir).expect("Could not chdir to temp test directory.");
 
-        let configs = Vec::<VsockConfig>::try_from(args);
-        assert!(configs.is_ok());
+        {
+            let socket_path = test_dir.path().join("vhost4.socket").display().to_string();
+            let uds_path = test_dir.path().join("vm4.vsock").display().to_string();
+            let args = VsockArgs::from_args(3, &socket_path, &uds_path, 64 * 1024);
 
-        let configs = configs.unwrap();
-        assert_eq!(configs.len(), 1);
+            let configs = Vec::<VsockConfig>::try_from(args);
+            assert!(configs.is_ok());
 
-        let config = &configs[0];
-        assert_eq!(config.get_guest_cid(), 3);
-        assert_eq!(config.get_socket_path(), "/tmp/vhost4.socket");
-        assert_eq!(config.get_uds_path(), "/tmp/vm4.vsock");
-        assert_eq!(config.get_tx_buffer_size(), 64 * 1024);
+            let configs = configs.unwrap();
+            assert_eq!(configs.len(), 1);
+
+            let config = &configs[0];
+            assert_eq!(config.get_guest_cid(), 3);
+            assert_eq!(config.get_socket_path(), socket_path);
+            assert_eq!(config.get_uds_path(), uds_path);
+            assert_eq!(config.get_tx_buffer_size(), 64 * 1024);
+        }
+
+        _ = test_dir.close();
     }
 
     #[test]
-    #[serial]
     fn test_vsock_config_setup_from_vm_args() {
-        let params = "--vm socket=/tmp/vhost3.socket,uds_path=/tmp/vm3.vsock \
```

Comment:
I honestly don't really like to change more than 500 lines just to indent by one block.

IIUC it's to clearly leave the creation of the temporary folder and chdir out of the test.
At this point how about a wrapper function for each test where we need to do it?

That way we minimize the changes.

Reply:
Removed, thanks!